src/main/scala/org/apache/spark/sql/kinesis/KinesisSourceRDD.scala

```diff
@@ -179,12 +198,14 @@ private[kinesis] class KinesisSourceRDD(
               // getShardIterator() should raise exception if its null if failOnDataLoss is true
               // if failOnDataLoss is false, getShardIterator() will be null and we should stop
               // fetching more records
+              addDelayInFetchingRecords(currentTimestamp)
```

Comment:
Address the same concerns which are addressed in the PR https://github.com/qubole/kinesis-sql/pull/65